### DISCUSSION TITLE
SOC estimation

### DISCUSSION THREAD

**Discussion by ddagrinho** [pybamm-team/PyBaMM#3438] [Q&A] [answered]
hello

i am using the parameter values of LFP(Prada2013), and as i can see the nominal discharge capacity is 1.1 Ah. although the averaged surface concentration is not 0% when i am using a current of 1.1 A for 1h.

Furthermore, the positive electrode is assumed to be LFP and the Negative electrode graphite. Although during discharge, the concentration of positive electrode is increased and the concentration of negative is decreased. LFP is the cathode material right? it should not be the opposite?

in the parameter from prada the initial concentration in the positive electrode(in the paper assumed to be lfp) is 35 mol/m3
and the initial concentration in the negative(graphite) is 30000 mol/m3
so i assume is completely discharged.although the current function that i apply is 1.1(which is a positive value).
![code2](https://github.com/pybamm-team/PyBaMM/assets/118671219/bdfcc39c-a3c6-44a5-b722-f4b70374b4aa)
![code1](https://github.com/pybamm-team/PyBaMM/assets/118671219/d5b9d79b-ec25-4399-95cc-215a860cce06)

**Comment by rtimms** [comment, marked as answer]
Hi, the Prada 2013 set is for a 2.3 Ah cell (see [this PR](https://github.com/pybamm-team/PyBaMM/pull/3096) where  the parameters in `Prada2013` were updated to better match those given in the paper, which is a 2.3 Ah cell, instead of the mix-and-match with the 1.1 Ah cell from Lain2019.)

During discharge, we expect the negative electrode to delithiate and the positive electrode to lithiate, so the negative concentration goes down and the positive concentration goes up. Also note that the electrodes won't be fully utilised, but operate in some stoichiometry window - at 0% SOC the negative electrode won't be totally empty and the positive electrode won't be totally full. There is a helper function in PyBaMM to do this, as shown in the example below.

```python3
import pybamm

model = pybamm.lithium_ion.DFN()
parameter_values = pybamm.ParameterValues("Prada2013")

x_0, x_100, y_100, y_0 = pybamm.lithium_ion.get_min_max_stoichiometries(
    parameter_values
)
print("x_0: ", x_0, "x_100: ", x_100, "y_100: ", y_100, "y_0: ", y_0)
experiment = pybamm.Experiment(["Discharge at 1 C until 2 V"])
sim = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment)
sim.solve()
print("Final time: " + str(sim.solution["Time [s]"].data[-1]) + " s")
sim.plot(
    [
        "Average negative particle concentration",
        "Average positive particle concentration",
        "Voltage [V]",
    ]
)

```

**Reply by ddagrinho** [reply]
okay i will check this. but then, during discharge, the positive electrode should be the graphite so as to increase the concentration. because in the documentation the negative electrode is the graphite in the parameter values.

**Comment by ddagrinho** [comment]
can somehow change these stoichiometric values so as to be able to reach closer to 0% and 100%?

**Reply by valentinsulzer** [reply]
You can see how changing `Q_n`, `Q_p`, or `Q_Li` affects the stoichiometry limits using this notebook: https://docs.pybamm.org/en/latest/source/examples/notebooks/models/electrode-state-of-health.html

`Q_n` and `Q_p` depend in electrode volume (height times width times thickness), active material volume fraction, and maximum particle concentration

`Q_Li` depends on `Q_n` and `Q_p` and initial stoichiometries.

In practice `Q_n`, `Q_p`, and `Q_Li` should be fitted to data e.g. using dVdQ analysis

**Reply by Thiyaga2025** [reply]
@valentinsulzer and @rtimms , thanks for the discussion. from this and other threads on stoichiometry related threads, i understand that x_0,x_100, y_0 and y_100 and calculated by bybamm using Vmax, Vmin, Q_cell, Q_p, and Q_n as detailed in electrode-state-of-health work sheet. in practice, when i fit a model in pybamm for a new cell, i can specify Vmax and Vmin directly in parameters. regarding Q values i understand from the above post that it can be altered by modifying electrode thickness, porosity, active material volume fraction and active material density. is my understanding correct. how can i check the values of all 'Q's before running the solution, i.e. the command to see the Q values. also, how maximum particle concentration enters the stoichiometry calculation as 'x' and 'y' values are already calculated by 'ElectrodeSOHSolver'.
thanks.
